nomt/benches/beatree.rs

```diff
@@ -1,7 +1,41 @@
-use criterion::{criterion_group, criterion_main, Criterion};
+#![cfg(feature = "benchmarks")]
 
-pub fn beatreee_benchmark(_c: &mut Criterion) {
-    todo!()
+use criterion::{criterion_group, criterion_main, BenchmarkId, Criterion};
+use nomt::{separate, Key};
+use rand::RngCore;
+
+// returns two keys a and b where b > a and b shares the first n bits with a
+fn get_keys(shared_bytes: usize) -> (Key, Key) {
+    let mut rand = rand::thread_rng();
+    let mut a = [0; 32];
+    rand.fill_bytes(&mut a[0..shared_bytes]);
+
+    // b > a
+    let mut b = a.clone();
+    b[shared_bytes] = 1;
+
+    (a, b)
+}
+
+fn separate_benchmark(c: &mut Criterion) {
+    let mut group = c.benchmark_group("separate");
+
+    for shared_bytes in 0..32 {
+        let input = get_keys(shared_bytes);
```

Comment:
better not to include generating keys in a benchmark, especially a microbenchmark. generate the data up-front, then benchmark it.